lib/src/intl/number_format.dart

```diff
@@ -618,9 +618,9 @@ class NumberFormat {
     if (simpleNumber < 100000000000000) return 14;
     if (simpleNumber < 1000000000000000) return 15;
     if (simpleNumber < 10000000000000000) return 16;
-    // We're past the point where being off by one on the number of digits
-    // will affect the pattern, so now we can use logs.
-    return max(1, (log(simpleNumber) / _ln10).ceil());
+    if (simpleNumber < 100000000000000000) return 17;
+    if (simpleNumber < 1000000000000000000) return 18;
+    return 19;
```

Comment:
I assume we're not expecting anything past 19?

Reply:
Anything larger cannot be represented in 64 bits, so we are safe!